### PR TITLE
Added smite reasons (updated, working)

### PR DESCRIPTION
**Pull request by MrImmaculate**
Adds reasons, makes them optional.

EDIT: issue https://github.com/TotalFreedom/TotalFreedomMod/issues/848

### CODE REVIEW

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 1]
```diff
@@ -28,14 +31,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        TFM_Util.bcastMsg(String.format("%s has been a naughty, naughty boy. \n" + ChatColor.YELLOW + "Reason: %s", player.getName(), reason), ChatColor.RED);
+
+        //Deop
+        player.setOp(false);
+
+        //Set gamemode to survival:
+        player.setGameMode(GameMode.SURVIVAL);
+
+        //Clear inventory:
+        player.getInventory().clear();
```
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Why are you clearing their inventory? I dont remember that ever being a thing...

**Comment by Wild1145** [comment]
This will now require smite to have an argument or will not work at all... 

Well... It kinda will, you will get the smite message but nothing will happen...

**Comment by DUFCLiam** [comment]
@Wild1145, It does clear your inventory here is the code for it: 
` //Clear inventory:
  player.getInventory().clear();`

**Comment by Wild1145** [comment]
@DUFCLiam I am aware that it is clearing the inventory, that is what I flagged. I dont see a productive reason as to why smite should wipe your inventory...

**Comment by DUFCLiam** [comment]
@Wild1145, I thought you where asking if they added that. It is in the base code of smiting and I do agree we can activate `gamerule keepInventory true` and that means we can remove all instances of clearing inventory.

**Comment by RobinGall2910** [comment]
@DUFCLiam it isn't that magical, there's several commands with that clearinventory code line, we might need to remove that first if you don't want the players inventory to get wiped.

**Comment by DUFCLiam** [comment]
I know, That's why I said `all instances`

**Comment by ImScoop** [comment]
@Wild1145 This works without a reason, and it smites. It was tested, maybe you should review it again.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 1 above)
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: @Wild1145 If you look at /smite's source, you can see that clearing inventories has always been a thing.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 2]
```diff
@@ -28,14 +31,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+            smite(player);
+        }
 
         return true;
     }
 
+    public static void smite(final Player player, final String reason)
+    {
+        // Smites the player with the reason under it in yellow
+        TFM_Util.bcastMsg(String.format("%s has been a naughty, naughty boy. \n" + ChatColor.YELLOW + "Reason: %s", player.getName(), reason), ChatColor.RED);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Duplicate code. Please use/modify the pre-existing method.

**Comment by JeromSar** [comment]
In order to have this PR merged. You'll need to make the change as mentioned above.
Additionally, all changes required for the original issue need to be resolved.

**Comment by ImScoop** [comment]
@JeromSar I have fixed it.

**Comment by ImScoop** [comment]
Wow, nevermind. I messed up big.

**Comment by ImScoop** [comment]
@JeromSar I would I merge this into one method while still making reasons optional? I am pretty sure it has to be split into two method for each instance.

**Comment by DUFCLiam** [comment]
Use "smite(sender_p);" after the first declaration.

**Comment by AvalancheYT** [comment]
@DUFCLiam, sender_p doesn't do anything (Correct me if I am mistaken).

**Comment by DUFCLiam** [comment]
It is because of the boolean as it does `Player sender_p` so it will make it like that; Also using `smite(sender_p);` will work because it will smite them due to `public static void smite`

**Comment by ImScoop** [comment]
@DUFCLiam No, that is incorrect.

**Comment by MrImmaculate** [comment]
I will take another attempt at combining both functions into one method by checking if the reason is null or not.

**Comment by MrImmaculate** [comment]
@JeromSar Updated.

**Comment by MrImmaculate** [comment]
Bump ;-; @JeromSar

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 3]
```diff
@@ -58,8 +77,19 @@ public static void smite(final Player player)
                 world.strikeLightning(strike_pos);
             }
         }
+      
 
         //Kill:
         player.setHealth(0.0);
+        
+        
+}
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Formatting,

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 4]
```diff
@@ -28,16 +31,32 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
 
-    public static void smite(final Player player)
-    {
-        TFM_Util.bcastMsg(player.getName() + " has been a naughty, naughty boy.", ChatColor.RED);
+        
 
+    public static void smite(final Player player, final String reason)
+    {
+        if (reason == null) {
+        TFM_Util.bcastMsg(player.getName() + " has been naughty, naughty boy.", ChatColor.RED);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 5]
```diff
@@ -28,16 +31,32 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        }
+        
+        else {
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 6]
```diff
@@ -28,16 +31,32 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        TFM_Util.bcastMsg(player.getName() + " has been a naughty, naughty boy.\n" + ChatColor.YELLOW + "Reason: " + reason, ChatColor.RED);
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 7]
```diff
@@ -58,8 +77,19 @@ public static void smite(final Player player)
...
 
         //Kill:
         player.setHealth(0.0);
+        
+        
+}
+
+   // Creates the null reason, this is done because of method recursion, so no reason can be given and smite(player); will run, keeping other commands using smite in tact.
+    static String reason = null;
+    public static void smite(final Player player)
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Formatting.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 8]
```diff
@@ -58,8 +77,19 @@ public static void smite(final Player player)
...
+      
 
         //Kill:
         player.setHealth(0.0);
+        
+        
+}
+
+   // Creates the null reason, this is done because of method recursion, so no reason can be given and smite(player); will run, keeping other commands using smite in tact.
+    static String reason = null;
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: This is not a good idea.

**Comment by JeromSar** [comment]
In order to have this PR merged. You'll need to make the change as mentioned above.
Additionally, all changes required for the original issue need to be resolved.

This means that other reasons (/stfu, /gtfo) must be added to. See issue https://github.com/TotalFreedom/TotalFreedomMod/issues/848

**Pull request closed without merging**